pkg/scheduler/framework/v1alpha1/framework.go

```diff
@@ -124,3 +163,35 @@ func (f *framework) RunUnreservePlugins(
 func (f *framework) NodeInfoSnapshot() *cache.NodeInfoSnapshot {
 	return f.nodeInfoSnapshot
 }
+
+func pluginNameToConfig(args []config.PluginConfig) map[string]*runtime.Unknown {
+	pc := make(map[string]*runtime.Unknown, 0)
+	for _, p := range args {
+		pc[p.Name] = &p.Args
+	}
+	return pc
+}
+
+func pluginsNeeded(plugins config.Plugins) map[string]struct{} {
```

Comment:
optional:
If you make this `map[string]bool` instead, then you can check `pg[pluginName]` directly without the `if _, ok := pg[pluginName]; ok {...` syntax.